run_common.go

```diff
@@ -2092,3 +2092,28 @@ func relabel(path, mountLabel string, shared bool) error {
 	}
 	return nil
 }
+
+// mapContainerNameToHostname returns the passed-in string with characters that
+// don't match the pattern "[A-Za-z0-9][A-Za-z0-9._-]+" stripped out.
+func mapContainerNameToHostname(containerName string) string {
+	const initialAllowedChars = "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ" + "0123456789"
```

Comment:
Would regex such as `^[A-Za-z0-9]` and `[A-Za-z0-9._-]` be easier to maintain ? or could it be slow here ?

Reply:
That's potentially a lot of calls to `FindStringIndex()`, but it's worth a shot if the result is more readable.